processor/k8sattributesprocessor/pod_association.go

```diff
@@ -91,6 +91,12 @@ func getConnectionIP(ctx context.Context) kube.PodIdentifier {
 	if c.Addr == nil {
 		return ""
 	}
+	switch addr := c.Addr.(type) {
```

Comment:
Are the cases here covered by unit tests?

Reply:
The existing case, net.IPAddr, is covered. I've added new tests for net.TCPAddr and net.UDPAddr